mlflow/store/artifact/azure_data_lake_artifact_repo.py

```diff
@@ -129,3 +140,103 @@ def _download_file(self, remote_file_path, local_path):
 
     def delete_artifacts(self, artifact_path=None):
         raise NotImplementedError("This artifact repository does not support deleting artifacts")
+
+    def _upload_to_cloud(self, cloud_credential_info, src_file_path, artifact_file_path):
```

Comment:
This function was inspired by the AWS upload implementation in DatabricksArtifactRepo to NOT do multipart upload if the file size is too small. I've also added an environment variable MLFLOW_ENABLE_MULTIPART_UPLOAD so that if users would not like to do multipart upload, they can opt-out.